### PR TITLE
Implement instruction JR NZ,r8 (0x20)

### PR DESCRIPTION
**Pull request by mcomstock**
Implemented the instruction, and manually tested that it works (both when the zero flag is set and unset). We need to address the issue of how to change the number of clock cycles for the instruction (see the comment).

Please be nice...this is my first PR.

### CODE REVIEW

File: src/cpu/instructions.rs [hunk 1]
```diff
@@ -105,6 +105,22 @@ impl super::Cpu {
             // NOP
             0x00 => (),
 
+            // JR NZ,r8
+            0x20 => {
+                if self.reg.f.contains(ZERO) {
```
Comment (euclio) [inline src/cpu/instructions.rs]: use `if !self.reg.f.contains(ZERO)` to remove else

File: src/cpu/instructions.rs [hunk 2]
```diff
@@ -105,6 +105,22 @@ impl super::Cpu {
...
+                    ()
+                }
+                else {
+                    let jump = instruction.operands()[0] as i8 as i16;
```
Comment (euclio) [inline src/cpu/instructions.rs]: Does just `as i16` not work here?

Comment (mcomstock) [reply src/cpu/instructions.rs]: I tried, and it doesn't work. It treats the `u8` as the last byte of a `u16`.

File: src/cpu/instructions.rs (hunk 1 above)
Comment (mcomstock) [reply src/cpu/instructions.rs]: Fixed

**Comment by mcomstock** [comment]
Fixed what I could.

File: src/cpu/instructions.rs (hunk 2 above)
Comment (euclio) [reply src/cpu/instructions.rs]: Gotcha. What about

```rust
let jump: i16 = instructions.operands[0].into();
```

Comment (euclio) [reply src/cpu/instructions.rs]: Type inference might be smart enough to leave off the `i16`

**Pull request merged**